modAL/models/base.py

```diff
@@ -249,16 +170,25 @@ def query(self, X_pool, *query_args, **query_kwargs) -> Union[Tuple, modALinput]
             Value of the query_strategy function. Should be the indices of the instances from the pool chosen to be
             labelled and the instances themselves. Can be different in other cases, for instance only the instance to be
             labelled upon query synthesis.
+            query_metrics: returns also the corresponding metrics, if return_metrics == True
         """
-        query_result = self.query_strategy(self, X_pool, *query_args, **query_kwargs)
 
-        if isinstance(query_result, tuple):
-            warnings.warn("Query strategies should no longer return the selected instances, "
-                          "this is now handled by the query method. "
-                          "Please return only the indices of the selected instances.", DeprecationWarning)
-            return query_result
+        try:
+            query_result, query_metrics = self.query_strategy(
+                self, X_pool, *query_args, **query_kwargs)
 
-        return query_result, retrieve_rows(X_pool, query_result)
+        except ValueError:
+            query_metrics = None
+            query_result = self.query_strategy(
+                self, X_pool, *query_args, **query_kwargs)
```

Comment:
Here, the function call and unpacking in the `try` block can throw other exceptions, not just `ValueError`. For instance, when running `modAL/examples/active_regression.py`, I have the following error:


```
Traceback (most recent call last):
  File "/home/tivadar/Projects/modAL/examples/active_regression.py", line 52, in <module>
    query_idx, query_instance = regressor.query(X)
  File "/home/tivadar/Projects/modAL/modAL/models/base.py", line 178, in query
    self, X_pool, *query_args, **query_kwargs)
TypeError: cannot unpack non-iterable numpy.int64 object
```

To circumvent this, I suggest to handle all exceptions with the `except` block. (This might not be the best practice ever.)